pkg/workload/workload.go

```diff
@@ -444,6 +443,15 @@ func HasAllChecksReady(wl *kueue.Workload) bool {
 	return true
 }
 
+// Returns true if all the mustHaveChecks are present in the workload.
+func HasAllChecks(wl *kueue.Workload, mustHaveChecks sets.Set[string]) bool {
+	mustHaveChecks = mustHaveChecks.Clone()
+	for i := range wl.Status.AdmissionChecks {
+		mustHaveChecks.Delete(wl.Status.AdmissionChecks[i].Type)
+	}
+	return mustHaveChecks.Len() == 0
```

Comment:
it should be equivalent to check that the lengths are the same and each type present in mustHaveChecks (because we know that all the types are different).

That should avoid one Clone

Reply:
Preemption will be a special kind of check that could be present in the workloads list but not in the queues. Hard-code ignoring it here will look strange and be hard to maintain.

Adding a fast way out in case the must have set is empty.

Reply:
> Preemption will be a special kind of check

@trasc Does that mean this?

https://github.com/kubernetes-sigs/kueue/blob/97bc58e555d47b395e8cedb57f56db4a9fec74f9/pkg/constants/constants.go#L41-L43

Reply:
yes